### PR TITLE
CI Check for `CMDGEN` blocks

### PR DESCRIPTION
**Pull request by HU90m**
- Renamed `DOCGEN` blocks to be more descriptive: `CMDGEN` blocks.
- Added CI check for `CMDGEN` blocks to ensure they stay up to date.

This is in preparation for moving from mdbook preprocessors to using `CMDGEN` blocks where possible,
allowing this documentation to be viewable in github or in plain text
as well as enabling the same link checks to be applied to generated content.

### CODE REVIEW

File: util/reggen/README.md
Comment (marnovandermaas) [inline util/reggen/README.md:99]: Where is this generated from? If you just added this into the readme, maybe mention the human readable name in the commit message.

File: hw/top_earlgrey/doc/design/README.md
Comment (marnovandermaas) [inline hw/top_earlgrey/doc/design/README.md:382]: Why do you prefer the name command generator as opposed to documentation generator?

File: util/cmdgen.py
Comment (marnovandermaas) [inline util/cmdgen.py:62]: Fixing the path in cmdgen is not mentioned in the commit message.

File: util/cmdgen.py
Comment (marnovandermaas) [inline util/cmdgen.py:136]: Maybe define what a glob is in the help message?

File: util/cmdgen.py
Comment (marnovandermaas) [inline util/cmdgen.py:24]: Capitalize "now" in commit message.

**Review by marnovandermaas — COMMENTED** [review]
Thanks for these changes, some initial comments form my side.

File: util/reggen/README.md
Comment (HU90m) [reply util/reggen/README.md:99]: These are generated using the command in the `util/selfdoc.py reggen` command. This optional field was added in [util/reggen/ip_block.py](https://github.com/lowRISC/opentitan/commit/6211349dbe2bde6626b19a4c44ed4f1c4d8a6896#diff-eab0f969150e7b8625e6753643a528484f1a6f07b71c2669e48e0f247e71d726).

I hadn't set up the CI check though, so the committer wasn't asked and told how to update the docs.

File: util/cmdgen.py
Comment (HU90m) [reply util/cmdgen.py:62]: This path is only used for log messages and doesn't effect functionality. I was adding log messages and thought the absolute paths were hurting readability.

File: hw/top_earlgrey/doc/design/README.md
Comment (HU90m) [reply hw/top_earlgrey/doc/design/README.md:382]: The name change from a discussion about DOCGEN being quite generic and not very descriptive. We have a lot of document generators: `rustdoc`, `doxygne`, `selfdoc`, etc.

I feel `CMDGEN` better describes what the tool does; It simply runs a command to generate content.

**Review by jwnrt — APPROVED** [review]
LGTM, though I didn't run it. I'm fine with the `DOCGEN -> CMDGEN` rename.

**Comment by HU90m** [comment]
Made suggested changes:
- Capitalised first character commit messages.
- Improved help message for positional arguments.
- Fixed module docstring typo

File: util/reggen/README.md
Comment (marnovandermaas) [reply util/reggen/README.md:99]: Ok, thanks

File: hw/top_earlgrey/doc/design/README.md
Comment (marnovandermaas) [reply hw/top_earlgrey/doc/design/README.md:382]: Ok, since @jwnrt seems to be happy with this change, I think we can keep it.

File: util/cmdgen.py
Comment (marnovandermaas) [reply util/cmdgen.py:62]: Ok, thanks

File: util/cmdgen.py
Comment (marnovandermaas) [reply util/cmdgen.py]: Yes, that's better thanks.

**Review by marnovandermaas — APPROVED** [review]
Thanks for making the clarifications. These changes look good to me. Let's wait for CI to finish before merging.

**Review by hcallahan-lowrisc — APPROVED** [review]
LGTM! It will be nice to have some proper documentation for this mechanism at some point, as I think it could be confusing for new devs / contributors. But looks great so far! Thanks @HU90m

**Comment by HU90m** [comment]
Rebased

File: util/cmdgen.py
Comment (hcallahan-lowrisc) [inline util/cmdgen.py:51]:
```suggestion
def cmdgen_check_file(filepath: Path, dry_run: bool, update: bool) -> bool:
```

File: util/cmdgen.py
Comment (hcallahan-lowrisc) [inline util/cmdgen.py:62]:
```suggestion
        bool: If neither dry_run or update are set, will return True if an update is required.
```

File: util/cmdgen.py
Comment (hcallahan-lowrisc) [inline util/cmdgen.py:135]: I don’t think the tool only applies to .md files with the globbing argument as-is? Is being more generic or more specific desirable here?
I think we should either 1) update the comment to not mention markdown specifically, or 2) add a check that excludes all non-markdown files from being updated. WDYT?

Comment (HU90m) [reply util/cmdgen.py]: I've just updated the comment for now.

**Comment by HU90m** [comment]
fixed lint failure

**Review by hcallahan-lowrisc — APPROVED** [review]

**Pull request merged**
